ale_linters/dockerfile/dockerfile_lint.vim

```diff
@@ -0,0 +1,66 @@
+" Author: Alexander Olofsson <alexander.olofsson@liu.se>
+
+call ale#Set('dockerfile_dockerfile_lint_executable', 'dockerfile_lint')
+call ale#Set('dockerfile_dockerfile_lint_options', '')
+
+function! ale_linters#dockerfile#dockerfile_lint#GetType(type) abort
+    if a:type is? 'error'
+        return 'E'
+    elseif a:type is? 'warn'
+        return 'W'
+    endif
+
+    return 'I'
+endfunction
+
+function! ale_linters#dockerfile#dockerfile_lint#Handle(buffer, lines) abort
+    try
+        let l:data = json_decode(join(a:lines, ''))
+    catch
+        return []
+    endtry
+
+    if empty(l:data)
+        " Should never happen, but it's better to be on the safe side
+        return []
+    endif
+
+    let l:messages = []
+
+    for l:type in ['error', 'warn', 'info']
+        for l:object in l:data[l:type]['data']
```

Comment:
Can any of these keys be missing? It might be best to use the `get()` function just in case.

Reply:
Well, if any of them are missing, then that's because dockerfile_lint has crashed completely when trying to lint the Dockerfile, which means the object won't be valid JSON either way.

Reply:
So, there will always be an `error`, `warn`, and `info` output, even if they're empty?

Reply:
If the the linter runs - and exits with a non-255 code, then it will spit out a JSON object that always looks like; (Though with data in place)
```json
{
  "error": {
    "count": 0,
    "data": []
  },
  "warn": {
    "count": 0,
    "data": []
  },
  "info": {
    "count": 0,
    "data": []
  },
  "summary": []
}
```

If - on the other hand - it exits with 255 then you'll get a JavaScript error backtrace as an output, which is not valid JSON and therefore won't reach the extraction part.

Reply:
Alright. In that case, we should be fine.